buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -350,14 +350,16 @@ class ClusterFormationTasks {
           the elasticsearch source tree then this should be the version of elasticsearch built by the source tree.
           If it isn't then Bad Things(TM) will happen. */
         Task extract = project.tasks.create(name: name, type: Copy, dependsOn: extractDependsOn) {
-            if (getOs().equals("windows")) {
-                from {
+            from {
+                File archiveFile = configuration.singleFile
+                if (archiveFile.getName().endsWith("zip")) {
                     project.zipTree(configuration.singleFile)
```

Comment:
Should this be using archiveFile?

Reply:
I just modified this section to be a direct porting of the change in your PR.